packet.go

```diff
@@ -132,27 +246,47 @@ func marshalIdString(packetType byte, id uint32, str string) ([]byte, error) {
 	return b, nil
 }
 
+func unmarshalIdString(b []byte, id *uint32, str *string) (err error) {
```

Comment:
can you return this string and the id, rather than updating the reference ?

please, no named return values unless absolutely necessary